swift-sdk/Internal/InternalIterableAPI.swift

```diff
@@ -158,6 +151,13 @@ final class InternalIterableAPI: NSObject, PushTrackerProtocol, AuthProvider {
         self._email = email
         self._userId = nil
         
+        if(config.enableAnonTracking) {
+            if(email != nil) {
+                attemptAndProcessMerge(merge: merge ?? true, replay: replay ?? true, destinationUser: email, isEmail: true, failureHandler: failureHandler)
+            }
+            self.localStorage.userIdAnnon = nil
+        }
+        
```

Comment:
```suggestion
        if config.enableAnonTracking, let email = email {
            attemptAndProcessMerge(
                merge: merge ?? true,
                replay: replay ?? true,
                destinationUser: email,
                isEmail: true,
                failureHandler: failureHandler
            )
            self.localStorage.userIdAnnon = nil
        }
```

Reply:
Just code style suggestion. It helps reducing the nesting and makes the code easier to read!
Also, in Swift you don't need to add the `( )` on if statements.

Reply:
Hmm, so maybe there's a race condition issue!